src/components/CartList.js

```diff
@@ -0,0 +1,73 @@
+import React, { useMemo } from "react";
+import {
+  CartContainer,
+  CartListContainer,
+  Item,
+  ProductDetails,
+  ProductImage,
+  ProductQuantity,
+  RemoveBtn,
+  Summary,
+} from "../styles/components/CartList.styles";
+import { Button } from "../styles/globalStyles";
+
+export const CartList = ({ items }) => {
+  const total = useMemo(
+    () => items.reduce((acum, curr) => acum + curr.price * curr.quantity, 0),
+    [items]
+  );
+  return (
+    <CartContainer>
+      <CartListContainer>
+        <h3>Shopping cart</h3>
+        <hr />
+        <table>
+          <thead>
+            <tr>
+              <th>Product details</th>
+              <th>Quantity</th>
+              <th>Price</th>
+              <th>Total</th>
+            </tr>
+          </thead>
+          <tbody>
+            {items.map((item) => (
+              <Item key={item.id}>
+                <ProductDetails>
+                  <ProductImage
+                    src={item.images[0]}
+                    alt={item.name}
+                  ></ProductImage>
+                  <div>
+                    <p>{item.name}</p>
+                    <p>Product code: {item.id}</p>
+                  </div>
+                </ProductDetails>
+                <td>
+                  <ProductQuantity>
+                    <input
+                      type="number"
+                      value={item.quantity}
+                      label="quantity"
+                      readOnly
+                    ></input>
```

Comment:
I cannot review this point because your code does not make modifications, it only reads the information from the store and this input does not have an `onChange={handleChange}`


> The "Quantity" input only allows positive numbers and affects the total cost of the product, the total items, and the total cost in the "Summary" section. (10 points)